libs/wingsdk/src/cloud/api.ts

```diff
@@ -0,0 +1,91 @@
+import { Construct } from "constructs";
+import { Polycons } from "polycons";
+import { FunctionProps } from "./function";
+import { Inflight, Resource } from "../core";
+import { Code } from "../core/inflight";
+
+/**
+ * Global identifier for `Api`.
+ */
+
+export const API_TYPE = "wingsdk.cloud.Api";
+
+/**
+ * Properties for `Api`.
+ */
+
+export interface ApiProps {}
+
+/**
+ * Functionality shared between all `Api` implementations.
+ */
+
+export abstract class ApiBase extends Resource {
+  public readonly stateful = true;
+  constructor(scope: Construct, id: string, props: ApiProps) {
+    super(scope, id);
+
+    if (!scope) {
+      return;
+    }
+
+    props;
+
+    this.display.title = "Api";
+
+    this.display.description = "A cloud API";
+  }
+  /**
+   * Add a function to handle GET requests to a route.
+   */
+  public abstract get(
+    route: string,
+    inflight: Inflight,
+    props?: ApiOnGetProps
+  ): void;
+}
+
+/**
+ * Options for Api.get.
+ */
+export interface ApiOnGetProps extends FunctionProps {}
+
+/**
+ * AWS infrastructure of `cloud.Api`.
+ */
+export class Api extends ApiBase {
+  constructor(scope: Construct, id: string, props: ApiProps = {}) {
+    super(null as any, id, props);
+    return Polycons.newInstance(API_TYPE, scope, id, props) as Api;
+  }
+
+  /** @internal */
+  public _toInflight(): Code {
+    throw new Error("Method not implemented.");
+  }
+
+  /**
+   *  Create a function to handle GET requests to a route.
+   *
+   * @param route
+   * @param inflight
+   * @param props
+   */
+  public get(route: string, inflight: Inflight, props?: ApiOnGetProps): void {
+    route;
+    inflight;
+    props;
+    throw new Error("Method not implemented.");
+  }
+}
+
+/**
+ * Inflight client for `IGetRequestHandler`.
+ */
+export interface IApiGetHandlerHandlerClient<ApiRequest, ApiResponse> {
```

Comment:
Why is this specific to "get"?

Reply:
It was the starting point with a lot of copy pasting from the bucket and queue implementation. I think you are right that this can be used for all http methods.